### PR TITLE
Generate the concourse keys using native Terraform resources

### PR DESCRIPTION
**Pull request by iuriaranda**
This commit changes the `keys` module to use native Terraform resources instead of commands through a local provisioner.
Makes it more reliable.

### CODE REVIEW

File: keys/variables.tf [hunk 1]
```diff
@@ -21,6 +16,11 @@ variable "concourse_workers_iam_role_arns" {
   description = "List of ARNs for the IAM roles that will be able to assume the role to access concourse keys in S3. Normally you'll include the Concourse worker IAM role here"
 }
 
+variable "generate_keys" {
```
Comment (SamClinckspoor) [inline keys/variables.tf]: You don't seem to be using this variable?

Comment (iuriaranda) [reply keys/variables.tf]: 😱 you're right. Good catch, I'll fix it

Comment (iuriaranda) [reply keys/variables.tf]: fixed

**Comment by so0k** [comment]
interesting, I used your TF as a guideline and had made the same changes

however, this leaves the keys in the TF state, while your previous code only ever kept them in encrypted in the s3 bucket, I thought your previous approach was better

**Comment by iuriaranda** [comment]
Hi @so0k , thanks for your feedback.

Indeed the new approach leaves the keys in the TF state, but that's why I made that functionality optional. In our case, all our Terraform state files are stored encrypted in S3, so we reduce the security risk. But if that's still an issue for you or other people, you can always turn off that feature by setting `generate_keys=false`, and generate and upload your keys outside of Terraform, or even via Terraform using the `local_file` provisioner approach outside of this module.

The idea behind this PR was to make it more generic and more usable to other people, by using standard Terraform resources and giving module users more options and flexibility.

**Pull request merged**